src/main/scala/cognite/spark/v1/DataModelInstancesRelation.scala

```diff
@@ -89,6 +89,45 @@ class DataModelInstanceRelation(config: RelationConfig, modelExternalId: String)
     )
   }
 
+  // scalastyle:off cyclomatic.complexity
+  def getInstanceFilter(sparkFilter: Filter): Option[DataModelInstanceFilter] =
+    sparkFilter match {
+      case EqualTo(left, right) => {
+        Some(DMIEqualsFilter(Seq(modelExternalId, left), parseValue(right)))
+      }
+      case In(attribute, values) =>
+        if (propertyTypes(attribute)._1.endsWith("[]")) {
+          None
+        } else {
+          val setValues = values.filter(_ != null)
+          Some(DMIInFilter(Seq(modelExternalId, attribute), setValues.map(parseValue)))
+        }
+      case GreaterThanOrEqual(attribute, value) =>
+        Some(DMIRangeFilter(Seq(modelExternalId, attribute), gte = Some(parseValue(value))))
+      case GreaterThan(attribute, value) =>
+        Some(DMIRangeFilter(Seq(modelExternalId, attribute), gt = Some(parseValue(value))))
+      case LessThanOrEqual(attribute, value) =>
+        Some(DMIRangeFilter(Seq(modelExternalId, attribute), lte = Some(parseValue(value))))
+      case LessThan(attribute, value) =>
+        Some(DMIRangeFilter(Seq(modelExternalId, attribute), lt = Some(parseValue(value))))
+      case And(f1, f2) =>
+        (getInstanceFilter(f1) ++ getInstanceFilter(f2)).reduceLeftOption((sf1, sf2) =>
+          DMIAndFilter(Seq(sf1, sf2)))
```

Comment:
I was writing a 4 case match case, then I thought there is probably a shortcut :D @exyi